Add feature flag to hide experimental Positron notebook editor

nstrayer opened the pull request:
Addresses #8600.

Adds `positron.notebook.enabled` configuration setting (default: false) to conditionally register the Positron notebook editor. When disabled, the Positron notebook option is hidden from the "Open With..." menu for .ipynb files, leaving only the default VS Code notebook editor available. When enabled, users can access the Positron notebook editor alongside the standard editor.

The setting is marked with the `experimental` tag and hidden from the Settings UI (`included: false`), making it a "secret" setting configurable only via JSON for advanced users. The setting name excludes "experimental" to allow future flexibility when the feature graduates from experimental status.

### Release Notes

#### New Features
- Added experimental setting to control availability of Positron notebook editor

#### Bug Fixes  
- N/A

### QA Notes

Verify the setting controls editor availability. Restart required for changes to take effect.

1. **Test default behavior (feature disabled):**
   - Right-click any `.ipynb` file → "Open With..."
   - Should only see standard VS Code notebook editor option

2. **Test enabled behavior:**
   - The setting is hidden from Settings UI and must be configured in JSON
   - Add to `settings.json`:
     ```json
     {
       "positron.notebook.enabled": true
     }
     ```
   - Restart Positron
   - Right-click any `.ipynb` file → "Open With..."
   - Should see both "Notebook Editor" and "Positron Notebook" options

3. **Test setting persistence:**
   - Verify the setting persists across Positron restarts
   - Verify disabling the setting removes the Positron notebook option

nstrayer committed:
notebooks: add feature flag to hide experimental Positron notebook editor

Add positron.notebook.experimental.enabled configuration setting (default: false)
to conditionally register the Positron notebook editor. When disabled, users will
use the default VS Code notebook editor for .ipynb files.

nstrayer committed:
tests: skip notebook editor tests while behind feature flag

Skip tests that expect Positron notebook editor to be available since
it's now behind positron.notebook.experimental.enabled (default: false).
Tests will be re-enabled when the feature flag is removed.

nstrayer committed:
config: hide experimental notebook setting from Settings UI

Add included: false to positron.notebook.experimental.enabled setting
to hide it from the Settings UI. The setting can only be configured
directly in the JSON settings file for advanced users.

nstrayer committed:
fix: improve configuration key consistency

- Add POSITRON_NOTEBOOK_EXPERIMENTAL_ENABLED_FULL_KEY constant for the full configuration path
- Use full key constant in affectsConfiguration() to avoid hardcoding
- Add clarifying comments to distinguish relative vs full configuration paths
- Addresses review feedback about configuration key inconsistency

nstrayer committed:
refactor: address review feedback on configuration design

- Use experimental tag instead of markdown for proper categorization
- Remove 'experimental' from setting name (positron.notebook.enabled) for future flexibility
- Update all constant names and function references accordingly
- Update test skip comments to reference new setting name
- Addresses feedback from @juliasilge about configuration best practices

nstrayer committed:
refactor: simplify configuration key constants

Remove unnecessary dual constants (POSITRON_NOTEBOOK_ENABLED_KEY and
POSITRON_NOTEBOOK_ENABLED_FULL_KEY) and use single constant with full
path everywhere.

juliasilge reviewed (dismissed):
Working as expected when not set or with `"positron.notebook.enabled": true` in my `settings.json`! 🙌

nstrayer committed:
remove: dead configuration change listener

Remove the configuration change listener that wasn't doing anything.
Since editor registration is one-way and can't be cleanly unregistered,
and a restart is required for changes anyway, the listener was just
dead code that gave the false impression of dynamic behavior.

Also remove the unused POSITRON_NOTEBOOK_ENABLED_KEY import.
